File: lib/auth/login_view_model.dart
import 'package:flutter/foundation.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:diplomovka/auth/auth_service.dart';
import 'package:diplomovka/auth/login_user_use_case.dart';

/// ViewModel pre Login obrazovku.
/// Drží stav (loading + error) a volá UseCase.
class LoginViewModel extends ChangeNotifier {
  final LoginUserUseCase _loginUserUseCase;

  LoginViewModel(AuthService authService)
      : _loginUserUseCase = LoginUserUseCase(authService);

  String? _error;
  bool _isLoading = false;

  String? get error => _error;
  bool get isLoading => _isLoading;

  /// Login, ktorý volá UI (LoginPage).
  Future<void> login(String email, String password) async {
    if (_isLoading) return;

    _isLoading = true;
    _error = null;
    notifyListeners();

    try {
      await _loginUserUseCase.execute(email, password);
      // úspech → _error ostáva null
    } on FirebaseAuthException catch (e) {
      // Tu mapuješ technické chyby na pekné texty pre užívateľa.
      if (e.code == 'user-not-found' || e.code == 'wrong-password') {
        _error = 'Neplatný e-mail alebo heslo.';
      } else if (e.code == 'invalid-email') {
        _error = 'Neplatný formát e-mailu.';
      } else if (e.code == 'user-disabled') {
        _error = 'Tento účet bol deaktivovaný.';
      }
      // 🚀 NOVÁ KONTROLA PRE OVERENIE E-MAILU
      else if (e.code == 'email-not-verified') {
        _error = e.message; // Použijeme správu, ktorú sme definovali v UseCase

        // Môžeme mu tu hneď poslať e-mail znova
        await AuthService().sendEmailVerification();
      }
      // ------------------------------------
      else {
        _error = 'Chyba pri prihlásení: ${e.message ?? e.code}';
      }
    } catch (e) {
      _error = 'Nastala neočakávaná chyba. Skúste to znova.';
      if (kDebugMode) {
        print('LoginViewModel error: $e');
      }
    } finally {
      _isLoading = false;
      notifyListeners();
    }
  }

  /// Vyčistí error – napr. keď zavrieš dialog, nech sa UI nefixluje na starú chybu.
  void clearError() {
    _error = null;
    notifyListeners();
  }
}

File: app/src/main/java/com/example/diplomovka_kotlin/viewmodel/LoginViewModel.kt
package com.example.diplomovka_kotlin.viewmodel

import android.content.Context
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.diplomovka_kotlin.data.services.AuthService
import com.example.diplomovka_kotlin.domain.LoginUserUseCase
import com.google.firebase.auth.FirebaseAuthException

class LoginViewModel(private val context: Context) : ViewModel() {

    private val authService = AuthService(context)
    private val loginUseCase = LoginUserUseCase(authService)

    private val _isLoading = MutableLiveData(false)
    val isLoading: LiveData<Boolean> = _isLoading

    private val _error = MutableLiveData<String?>(null)
    val error: LiveData<String?> = _error

    suspend fun login(email: String, password: String): Boolean {
        if (_isLoading.value == true) return false
        _isLoading.value = true
        _error.value = null

        return try {
            loginUseCase.execute(email, password)
            true
        } catch (e: FirebaseAuthException) {
            _error.value = when (e.errorCode) {
                "user-not-found", "wrong-password" -> "Neplatný e-mail alebo heslo."
                "invalid-email"                    -> "Neplatný formát e-mailu."
                "user-disabled"                    -> "Tento účet bol deaktivovaný."
                "email-not-verified"               -> {
                    authService.sendEmailVerification()
                    e.message
                }
                else -> "Chyba pri prihlásení: ${e.message ?: e.errorCode}"
            }
            false
        } catch (e: Exception) {
            _error.value = "Nastala neočakávaná chyba. Skúste to znova."
            false
        } finally {
            _isLoading.value = false
        }
    }

    fun clearError() { _error.value = null }
}
